examples/generative/corrdiff/datasets/hrrrmini.py

```diff
@@ -0,0 +1,198 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import json
+import math
+from typing import List, Tuple, Union
+
+import cftime
+import numpy as np
+from numba import jit, prange
+import xarray as xr
+
+from .base import ChannelMetadata, DownscalingDataset
+
+
+class HRRRMiniDataset(DownscalingDataset):
```

Comment:
Please add API documentation here, as well as a general documentation in the README.

Reply:
Also, what do you think about moving this to the Modulus datapipes directory?

Reply:
I would suggest keeping it as is for now - I'd support moving the`DownscalingDataset` base class to Modulus though, once the API becomes stable. At least the lat/lon handling should be figured out before I'd consider it stable.